components/pagination/pagination.ts

```diff
@@ -0,0 +1,163 @@
+import { Component, Input, Output, OnChanges, SimpleChanges, EventEmitter } from "@angular/core";
+
+@Component({
+    selector: "sui-pagination",
+    template: `
+<div class="ui pagination menu">
+    <a *ngIf="showBoundary" class="item"  (click)="selectPage(1)" [ngClass]="{'disabled': page===1}">
+        <i class="fast backward icon"></i>
+    </a>
+    <a *ngIf="showNavigation" class="item" (click)="selectPage(page-1)" [ngClass]="{'disabled': !hasPrevious()}">
+        <i class="step backward icon"></i>
+    </a>
+    <a *ngFor="let p of pages" class="item" [ngClass]="{'active':p==page}" (click)="page=p">
+        {{ p }}
+    </a>
+    <a *ngIf="showNavigation" class="item" (click)="selectPage(page+1)" [ngClass]="{'disabled': !hasNext()}">
+        <i class="step forward icon"></i>
+    </a>
+    <a *ngIf="showBoundary" class="item"  (click)="selectPage(pageCount)" [ngClass]="{'disabled': page===pageCount}">
+        <i class="fast forward icon"></i>
+    </a>
+</div>
+`
+})
+export class SuiPagination implements OnChanges {
+
+    // Public members
+    public pages:number[] = [];
+    public pageCount:number;
+
+    @Output()
+    public pageChanged:EventEmitter<number> = new EventEmitter<number>();
```

Comment:
Please also rename this `pageChange`, so `[(page)]` syntax can be used (see below)